Inferences/InductionHelper.cpp

```diff
@@ -209,22 +208,61 @@ bool InductionHelper::isInductionTermFunctor(unsigned f) {
          );
 }
 
+bool containsSkolem(Term* t) {
+  if (env.signature->getFunction(t->functor())->skolem()) return true; 
+  NonVariableIterator nvi(t);
+  while (nvi.hasNext()) {
+    if (env.signature->getFunction(nvi.next().term()->functor())->skolem()) return true;
+  }
+  return false;
+}
+
+bool termAndLiteralSatisfyStrictness(const TermList& tl, Literal* l, unsigned strictness) {
```

Comment:
Could we introduce a "strictness" enumeration and/or make this a `switch` statement? It seems like there are only a finite number of permitted values with fixed semantics.

Reply:
This is a good question and I'm not sure what's the best solution. This function only checks a part of the strictness condition - for either equality or comparison.
There are 5 permitted values for equalities, (the same) 5 for comparisons, and 3 for terms, so altogether it's 75. I think having 75 values for this option is not really practical.
Another possibility would be to keep the option as an unsigned value, but then convert its sub-values for equalities, comparisons and terms into enums. I'm not sure though whether this would be much better than what we have now (converting the sub-values into unsigneds).
The last possibility I see would be to split this option into 3 options, each one an enum, but that would (unnecessarily, seems to me) add 3 options when 1 is sufficient.
What do you think?

Reply:
OK, I hoped that the 'strictness' semantics for equalities/comparisons/terms overlapped, but looking closer I don't think so. Then I think this makes sense and is an elegant solution: Martin might have some ideas about how to ensure that all invalid values for the option are prevented (e.g. `66`) as he's been improving the options machinery recently.